operator/pkg/helmreconciler/reconciler.go

```diff
@@ -291,33 +267,88 @@ func overallStatus(componentStatus map[string]*v1alpha1.InstallStatus_VersionSta
 	return ret
 }
 
-// allObjectHashes returns a map with object hashes of all the objects contained in cmm as the keys.
-func allObjectHashes(m string) map[string]bool {
-	ret := make(map[string]bool)
-	objs, err := object.ParseK8sObjectsFromYAMLManifest(m)
+// getOwnerLabels returns a map of labels for the given component name, revision and owning CR resource name.
+func (h *HelmReconciler) getOwnerLabels(componentName string) (map[string]string, error) {
+	crName, err := h.getCRName()
 	if err != nil {
-		scope.Error(err.Error())
+		return nil, err
 	}
-	for _, o := range objs {
-		ret[o.Hash()] = true
+	labels := make(map[string]string)
+	revision := ""
+	if h.iop != nil {
+		revision = h.iop.Spec.Revision
 	}
 
-	return ret
+	// Only pilot component uses revisions
```

Comment:
Why only pilot, ingress installed with version 1.6.0, will istiod with version old, that cause upgrade to fail

https://github.com/istio/istio/issues/23923